### PR TITLE
Remove start_event_loop_default.  Let pause() run the event loop for all backends.

### PR DESCRIPTION
**Pull request by anntzer**
### First commit: Move start_event_loop_default to start_event_loop.
There's no benefit to the end user to having a deprecation warning in
start_event_loop (which gets called e.g. by ginput) -- it's essentially
just there to let the devs know that they could implement a more
efficient (GUI toolkit-dependent) version.  So just provide the default
implementations instead and add a note that GUI toolkits may override
them.

flush_events doesn't need to raise by default, it can just return
nothing for non-interactive backends.

### Second commit: Let pause always run the event loop.
The default `start_event_loop` is not as efficient as it could be for
non-interactive backends (it runs a busy loop) but if you really care
you should just call time.sleep in that case.

Meanwhile, this avoids the need for third-party interactive backends to
register themselves into rcsetup.interactive_bk in order to be correctly
pause()able.

### CODE REVIEW

File: lib/matplotlib/backend_bases.py [hunk 1]
```diff
@@ -2368,55 +2368,24 @@ def new_timer(self, *args, **kwargs):
...
+    def start_event_loop(self, timeout=0):
+        """Start a blocking event loop.
 
-    def stop_event_loop(self):
-        """
-        Stop an event loop.  This is used to stop a blocking event
-        loop so that interactive functions, such as ginput and
-        waitforbuttonpress, can wait for events.
+        Such an event loop is used by interactive functions, such as ginput and
+        waitforbuttonpress, to wait for events.  This should not be confused
```
Comment (tacaswell) [inline lib/matplotlib/backend_bases.py]: I do not think this is correct, the GUI event loop is not always running.

Comment (anntzer) [inline lib/matplotlib/backend_bases.py]: Feel free to edit the message and force push if you'd like to.

**Comment by tacaswell** [comment]
@anntzer force-pushed to this branch.

**Comment by anntzer** [comment]
Rebased.

**Comment by anntzer** [comment]
Marking this as release critical only on the basis of "it would be nice to have it for mpl_cairo (specifically not needing to register itself into interactive_bk), and it's already been approved once" -- so just hoping to catch another core dev's attention.  Feel free to untag if you disagree.

**Comment by efiring** [comment]
I will leave this for @tacaswell to decide whether to backport for 2.1.

**Comment by tacaswell** [comment]
Backported to v2.1.x as 35af6c02c

**Pull request merged**